pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -60,6 +60,12 @@ func (r *pullthroughBlobStore) remoteStat(ctx context.Context, dgst digest.Diges
 		return distribution.Descriptor{}, err
 	}
 
+	r.pullFromInsecureRegistries = false
```

Comment:
An incorrect value can be set upon creation. That is why the `TestPullthroughServeBlob` failed.